experimental/dds/tree2/src/test/snapshots/summary.spec.ts

```diff
@@ -0,0 +1,117 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import path from "path";
+import { ISummaryTreeWithStats } from "@fluidframework/runtime-definitions";
+import { TestTreeProviderLite } from "../utils";
+import { brand, useDeterministicStableId } from "../../util";
+import {
+	FieldKey,
+	UpPath,
+	rootFieldKey,
+	fieldSchema,
+	GlobalFieldKey,
+	SchemaData,
+} from "../../core";
+import { ISharedTree, ISharedTreeView } from "../../shared-tree";
+import { singleTextCursor, FieldKinds, namedTreeSchema } from "../../feature-libraries";
+import { createSnapshot, isEqualPastSnapshot } from "./utils";
+
+const dirPathTail = "src/test/snapshots/files";
+const fieldKeyA: FieldKey = brand("FieldA");
+const fieldKeyB: FieldKey = brand("FieldB");
+const fieldKeyC: FieldKey = brand("FieldC");
+
+const globalFieldKey: GlobalFieldKey = brand("globalFieldKey");
+const rootFieldSchema = fieldSchema(FieldKinds.value);
+const globalFieldSchema = fieldSchema(FieldKinds.value);
+const rootNodeSchema = namedTreeSchema({
+	name: brand("TestValue"),
+	extraLocalFields: fieldSchema(FieldKinds.sequence),
+	globalFields: [globalFieldKey],
+});
+const testSchema: SchemaData = {
+	treeSchema: new Map([[rootNodeSchema.name, rootNodeSchema]]),
+	globalFieldSchema: new Map([
+		[rootFieldKey, rootFieldSchema],
+		[globalFieldKey, globalFieldSchema],
+	]),
+};
+
+function generateTree(fields: FieldKey[], height: number, nodesPerField: number): ISharedTree {
```

Comment:
IMO this would read more cleanly by converting these to named parameters:

```suggestion
function generateTree({ fields, height, nodesPerField }: { fields: FieldKey[], height: number, nodesPerField: number }): ISharedTree {
```

and updating call-sites